pkg/etcd/etcd.go

```diff
@@ -1162,46 +1194,64 @@ func (e *ETCD) trackLearnerProgress(ctx context.Context, progress *learnerProgre
 	return e.setLearnerProgress(ctx, progress)
 }
 
-func (e *ETCD) setEtcdStatusCondition(node *v1.Node, client corev1.CoreV1Interface, memberName string, promoted bool) (*v1.Node, []byte, error) {
-	etcdStatusType := v1.NodeConditionType("EtcdIsVoter")
+func (e *ETCD) getETCDStatus(ctx context.Context, url string) (*clientv3.StatusResponse, error) {
+	ctx, cancel := context.WithTimeout(ctx, defaultDialTimeout)
+	defer cancel()
+
+	return e.client.Status(ctx, url)
+}
 
+func (e *ETCD) setEtcdStatusCondition(node *v1.Node, client kubernetes.Interface, memberName string, memberStatus MemberStatus) error {
 	var newCondition v1.NodeCondition
-	if promoted {
+	switch memberStatus {
+	case StatusLearner:
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "False",
+			Reason:  "MemberIsLearner",
+			Message: "Node has not been promoted to voting member of the etcd cluster",
+		}
+	case StatusVoter:
 		newCondition = v1.NodeCondition{
 			Type:    etcdStatusType,
 			Status:  "True",
 			Reason:  "MemberNotLearner",
 			Message: "Node is a voting member of the etcd cluster",
 		}
-	} else {
+	case StatusUnhealthy:
 		newCondition = v1.NodeCondition{
 			Type:    etcdStatusType,
 			Status:  "False",
-			Reason:  "MemberIsLearner",
-			Message: "Node has not been promoted to voting member of the etcd cluster",
+			Reason:  "Unhealthy",
+			Message: "Node is unhealthy",
+		}
+	case StatusUnjoined:
+		newCondition = v1.NodeCondition{
+			Type:    etcdStatusType,
+			Status:  "False",
+			Reason:  "NotAMember",
+			Message: "Node is not a member of the etcd cluster",
```

Comment:
Add a default at the end that logs a warning or something, just to cover all the cases

Reply:
If it's an unknown status, should we change the status condition or should we just log it?

Reply:
I'd say just log it and return, since it's a developer problem and not something the user can do something about.